// @livekit/rtc-node/dist/ffi_client.js
import EventEmitter from 'events';
import { FfiHandle, livekitCopyBuffer, livekitDispose, livekitFfiRequest, livekitInitialize, livekitRetrievePtr, } from './napi/native.js';
import { FfiEvent, FfiRequest, FfiResponse } from './proto/ffi_pb.js';
export { FfiHandle, FfiEvent, FfiResponse, FfiRequest, livekitDispose as dispose };
export var FfiClientEvent;
(function (FfiClientEvent) {
    FfiClientEvent["FfiEvent"] = "ffi_event";
})(FfiClientEvent || (FfiClientEvent = {}));
export class FfiClient extends EventEmitter {
    /** @internal */
    static get instance() {
        if (!FfiClient._client)
            FfiClient._client = new FfiClient();
        return FfiClient._client;
    }
    constructor() {
        super();
        this.setMaxListeners(0);
        livekitInitialize((event_data) => {
            const event = FfiEvent.fromBinary(event_data);
            this.emit(FfiClientEvent.FfiEvent, event);
        }, true);
    }
    request(req) {
        const request = new FfiRequest(req);
        const req_data = request.toBinary();
        const res_data = livekitFfiRequest(req_data);
        return FfiResponse.fromBinary(res_data).message.value;
    }
    copyBuffer(ptr, len) {
        return livekitCopyBuffer(ptr, len);
    }
    retrievePtr(data) {
        return livekitRetrievePtr(data);
    }
    async waitFor(predicate) {
        return new Promise((resolve) => {
            const listener = (ev) => {
                if (predicate(ev)) {
                    this.off(FfiClientEvent.FfiEvent, listener);
                    resolve(ev.message.value);
                }
            };
            this.on(FfiClientEvent.FfiEvent, listener);
        });
    }
}
